integration/actions/pipeline_test.go

```diff
@@ -0,0 +1,119 @@
+//go:build actions
+// +build actions
+
+// Copyright 2022 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package actions
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/okteto/okteto/integration"
+	"github.com/okteto/okteto/pkg/model"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	deployPipelinePath  = "okteto/pipeline"
+	destroyPipelinePath = "okteto/destroy-pipeline"
+
+	githubURL = "https://github.com"
+
+	pipelineRepo    = "okteto/movies"
+	pipelineRepoURL = "git@github.com:okteto/movies.git"
+	pipelineFolder  = "movies"
+)
+
+func TestPipelineActions(t *testing.T) {
+	integration.SkipIfWindows(t)
+
+	namespace := getTestNamespace()
+
+	assert.NoError(t, executeCreateNamespaceAction(namespace))
+	assert.NoError(t, executeDeployPipelineAction(t, namespace))
+	assert.NoError(t, executeDestroyPipelineAction(namespace))
+	assert.NoError(t, executeDeleteNamespaceAction(namespace))
+}
+
+func executeDeployPipelineAction(t *testing.T, namespace string) error {
+	actionRepo := fmt.Sprintf("%s%s.git", githubHTTPSURL, deployPipelinePath)
+	actionFolder := strings.Split(deployPipelinePath, "/")[1]
+	log.Printf("cloning pipeline repository: %s", actionRepo)
+	if err := integration.CloneGitRepoWithBranch(actionRepo, oktetoVersion); err != nil {
+		if err := integration.CloneGitRepo(actionRepo); err != nil {
+			return err
+		}
+		log.Printf("cloned repo %s main branch\n", actionRepo)
+	}
+	log.Printf("cloned repo %s \n", actionRepo)
+	defer integration.DeleteGitRepo(actionFolder)
+
+	t.Setenv(model.GithubRepositoryEnvVar, pipelineRepo)
```

Comment:
This test still relies on movies repo. Any change on the movies repo could make this tests fail. Mentioning just to highlight that we still have some dependency for this repo. For example, this repo would be broken if we remove the `master` branch again from the remote

Reply:
Pipeline and preview commands needs a URL link to deploy the application. Those are the only two commands that relies on an okteto repo. Our most stable sample is movies, this is way it's used.

Reply:
it is the most stable repo but this week tests were broken because we have removed a remote branch. It is even using an old version of the movies app not the one we are actively using and maintaining.

I would say to create a fork or another repository called `e2e test application` or similar and start using that one. Documenting it properly in the README